### PR TITLE
multi-label support

### PR DESCRIPTION
**Pull request by tsachiherman**

### CODE REVIEW

File: util/metrics/counter.go [hunk 1]
```diff
@@ -194,12 +194,20 @@ func (counter *Counter) AddMetric(values map[string]string) {
...
+			}
 
-		values[counter.name] = strconv.FormatFloat(sum, 'f', -1, 32)
+			values[counter.name] = strconv.FormatFloat(sum, 'f', -1, 32)
+		}
+	} else {
+		for label, counterIdx := range counter.labels {
+			valueIdx := counter.valuesIndices[counterIdx]
+			value := counter.values[valueIdx]
+			values[counter.name+"["+label+"]"] = strconv.FormatFloat(value.counter, 'f', -1, 32)
```
Comment (cce) [inline util/metrics/counter.go:210]: These names end up as JSON field keys in the heartbeat, and any string is a valid key, but I think this is as fine an encoding as any.

**Comment by cce** [comment]
I modified @brianolson's benchmark from #1938 to use `Counter.Inc` with pre-created labels `map[string]string{"tag": "val"}` instead of `TagCounter.Add` and the difference was stark .. 19ns/op vs 5460 ns/op for using Counter.Inc

**Comment by cce** [comment]
Passing Counter.Inc(nil) brought it down to 7.6 ns/op which is in line with the comment "Much faster if labels is nil or empty."

**Review by cce — APPROVED** [review]

**Review by cce — CHANGES_REQUESTED** [review]
I wonder if it makes more sense to convert the handful of remaining users of 1-D labels with this to use TagCounters instead?

**Comment by tsachiherman** [comment]
> I wonder if it makes more sense to convert the handful of remaining users of 1-D labels with this to use TagCounters instead?

I would not mind if we do both.

**Comment by algorandskiy** [comment]
@cce maybe take over it, amend with the suggestion and merge in?

**Pull request closed without merging**